runner_support.js

```diff
@@ -157,23 +138,33 @@ exports.runTests = function runTests(runner, key, family, options) {
                               '}\n';
                 }
                 if (/\b__createIterableObject\b/.test(evalcode)) {
-                    script += exports.createIterableHelper;
+                    script += createIterableHelper;
                 }
 
+                script += `var testCode = ${JSON.stringify(evalcode)};`;
+
                 if (/\basyncTestPassed\b/.test(evalcode)) {
-                    script += asyncTestHelperHead + '\n' + evalcode + '\n\n' + asyncTestHelperTail;
+                    script += `
+                        try {
+                            ${asyncTestHelperHead}
+                            eval(testCode);
+                            ${asyncTestHelperTail}
+                        } catch (e) {};`;
                 } else if (/\bglobal\.test\b/.test(evalcode)) {
-                    script += 'global.test = function (expression) {\n' +
-                              '    if (expression) {\n' +
-                              '        console.log("[SUCCESS]");\n' +
-                              '    }\n' +
-                              '}\n' +
-                              evalcode;
+                    script += `
+                        global.test = function (expression) {
+                            if (expression) ${logCommand}("[SUCCESS]");
```

Comment:
let's keep the curly braces; don't want another gotofail :-)
```suggestion
                            if (expression) {
                                ${logCommand}("[SUCCESS]");
                            }
```

Reply:
done